### PR TITLE
doc: Add three new glossaries

### PR DESCRIPTION
**Pull request by lirui34**
Add three new glossaries: "Hybrid Mode", "Partition Mode", "Sharing Mode".

Signed-off-by: lirui34 <ruix.li@intel.com>

### CODE REVIEW

**Comment by hwang37** [comment]
Right. We'll add additional terms in separate PR. For example "SDC-Basic" or "SDV-Basic" after receiving final confirmation.

File: doc/glossary.rst [hunk 1]
```diff
@@ -86,6 +86,10 @@ Glossary of Terms
    High GM
       See :term:`Hidden GM`
 
+   Hybrid Mode
+      One of three operation modes which ACRN can support. The mixed mode
```
Comment (deb-intel) [inline doc/glossary.rst]:
```suggestion
      One of three operation modes (hybrid, partition, sharing) that ACRN supports. 
```

File: doc/glossary.rst [hunk 2]
```diff
@@ -86,6 +86,10 @@ Glossary of Terms
    High GM
       See :term:`Hidden GM`
 
+   Hybrid Mode
+      One of three operation modes which ACRN can support. The mixed mode 
+      of supporting both partitioned user VM and sharing user VMs
```
Comment (deb-intel) [inline doc/glossary.rst]:
```suggestion
In this mixed mode, physical hardware resources can be both partitioned to individual user VMs and shared across user VMs.
```

File: doc/glossary.rst [hunk 3]
```diff
@@ -124,6 +128,10 @@ Glossary of Terms
       the Project ACRN architecture, pass-through devices are owned by the
       foreground OS.
 
+   Partition Mode
+      One of three operation modes which ACRN can support. Physical
```
Comment (deb-intel) [inline doc/glossary.rst]:
```suggestion
      One of three operation modes (partition, sharing, hybrid) that ACRN supports. 
```

File: doc/glossary.rst [hunk 4]
```diff
@@ -124,6 +128,10 @@ Glossary of Terms
       the Project ACRN architecture, pass-through devices are owned by the
       foreground OS.
 
+   Partition Mode
+      One of three operation modes which ACRN can support. Physical 
+      hardware resources are partitioned to individual user VM.
```
Comment (deb-intel) [inline doc/glossary.rst]:
```suggestion
      Physical hardware resources are partitioned to individual user VMs.
```

File: doc/glossary.rst [hunk 5]
```diff
@@ -180,6 +188,10 @@ Glossary of Terms
       access hardware resources directly by running native drivers and
       provides device sharing services to User VMs via the Device Model.
 
+   Sharing Mode
+      One of three operation modes which ACRN can support. Most of physical
```
Comment (deb-intel) [inline doc/glossary.rst]:
```suggestion
      One of three operation modes (sharing, hybrid, partition) that ACRN supports. Most of the physical 
```

**Pull request merged**